libchihua-msg/src/node.c

```diff
@@ -0,0 +1,81 @@
+#include "../include/node.h"
+#include "../include/node/peer-dbus.h"
+#include "./common/memory.h"
+
+#include <stdio.h>
+
+Node *node_new(const NodeParams *p) {
+        fprintf(stdout, "Creating Node...\n");
+
+        int r = 0;
+        sd_event *event = NULL;
```

Comment:
```_cleanup_free_``` should be used here so that GCC performs automatic cleanup of any memory that gets allocated for this object if it goes out of scope in the error case.  Note that it's the use of ```_cleanup_free_``` here that is the reason for using ```steal_pointer()``` below.

Reply:
Good catch! I forgot about the cleanup here. Instead of `_cleanup_free_` we need to add [\_cleanup_sd_event\_](https://github.com/engelmi/orch/blob/f23f361684bd00ada9465a144c61ed769b788bc4/libchihua-msg/src/common/dbus.h#L9) so `sd_event_unrefp` is called, I think.